challenge_demo/test/grammar_test.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import print_function
+import collections
+import unittest
+
+from grammar_parser.cfgparser import CFGParser
+from robocup_knowledge import load_knowledge
+from robot_skills.mockbot import Mockbot
+from action_server.actions.action import ConfigurationResult
+from action_server.task_manager import TaskManager
+
+TMTestResult = collections.namedtuple("TMTestResult", ["recipe", "config_result"])
+
+
+class GrammarTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.knowledge = load_knowledge('challenge_demo')
+        cls.parser = CFGParser.fromstring(cls.knowledge.grammar)
+        cls.task_manager = TaskManager(robot=Mockbot())
+
+        # ToDo: Should we make everything recursive???
+
+        # Rule corresponding to the grammar target
+        root_rule = cls.parser.rules[cls.knowledge.grammar_target]
+
+        # In principle, a list of actions can be provided to the actions server
+        # Therefore, we need to take to steps to list all actions
+        # I.e., the following actions_list_target refers to the grammar target
+        # containing a list of actions
+        actions_list_target = ""
+        for option in root_rule.options:
+            if "actions" not in option.lsemantic:
+                continue
+            actions_list_target = option.conjuncts[0].name  # This is too hardcoded
+            break
```

Comment:
As far as I see it the action_list_target is only a single string right? So either it is not a list, or the list is within the string?

Reply:
It's the 'target', i.e., the string identifying the underlying list of actions in the grammar